src/domain/wallet-invoices/wallet-invoice-factory.types.d.ts

```diff
@@ -0,0 +1,8 @@
+type WalletInvoiceFactoryType = {
```

Comment:
Don't think you need the post-fix `Type`. Something with that post fix implies it parameterises another type like:
```
type SomeErrorType
type Error<T>
type SpecificError = Error<SomeErrorType>
```